### PR TITLE
WIP: Fix Python (Pt. 2)

### PR DESCRIPTION
**Pull request by jakirkham**
This is an attempt to fix some other Python issues. It sort of works, but OpenCV seems to keep trying to do both Python 2 and Python 3 builds. 😖 Though maybe you have some other ideas of how to fix it.

xref: https://github.com/conda-forge/opencv-feedstock/pull/16

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -5,20 +5,21 @@ SHORT_OS_STR=$(uname -s)
...
     OPENMP="-DWITH_OPENMP=1"
+    OCV_PYTHON=""
 fi
 if [ "${SHORT_OS_STR}" == "Darwin" ]; then
     DYNAMIC_EXT="dylib"
     OPENMP=""
+    OCV_PYTHON="-DCMAKE_FIND_FRAMEWORK=LAST"
 fi
 
-INC_PYTHON="${PREFIX}/include/python${PY_VER}"
```
Comment (jakirkham) [inline recipe/build.sh:14]: This was a horrible idea. So, it had to be removed.

File: recipe/build.sh [hunk 2]
```diff
@@ -5,20 +5,21 @@ SHORT_OS_STR=$(uname -s)
 if [ "${SHORT_OS_STR:0:5}" == "Linux" ]; then
     DYNAMIC_EXT="so"
     OPENMP="-DWITH_OPENMP=1"
+    OCV_PYTHON=""
 fi
 if [ "${SHORT_OS_STR}" == "Darwin" ]; then
     DYNAMIC_EXT="dylib"
     OPENMP=""
+    OCV_PYTHON="-DCMAKE_FIND_FRAMEWORK=LAST"
```
Comment (jakirkham) [inline recipe/build.sh:13]: Otherwise it picks up the system Python. 😱

File: recipe/build.sh [hunk 3]
```diff
@@ -5,20 +5,21 @@ SHORT_OS_STR=$(uname -s)
...
 fi
 
-INC_PYTHON="${PREFIX}/include/python${PY_VER}"
 if [ $PY3K -eq 1 ]; then
     PY_VER_M="${PY_VER}m"
     LIB_PYTHON="${PREFIX}/lib/libpython${PY_VER_M}.${DYNAMIC_EXT}"
-    OCV_PYTHON="-DBUILD_opencv_python3=1 -DPYTHON3_EXECUTABLE=${PYTHON} -DPYTHON3_INCLUDE_DIR=${PREFIX}/include/python${PY_VER_M} -DPYTHON3_LIBRARY=${PREFIX}/lib/libpython${PY_VER_M}.${DYNAMIC_EXT} -DPYTHON3_PACKAGES_PATH=${SP_DIR}"
+    OCV_PYTHON="${OCV_PYTHON} -DBUILD_opencv_python2=0 -DBUILD_opencv_python3=1 -DPYTHON3_EXECUTABLE=${PYTHON} -DPYTHON3_INCLUDE_DIR=${PREFIX}/include -DPYTHON3_LIBRARY=${PREFIX}/lib/libpython${PY_VER_M}.${DYNAMIC_EXT} -DPYTHON3_PACKAGES_PATH=${SP_DIR}"
```
Comment (jakirkham) [inline recipe/build.sh:19]: Amazingly `-DBUILD_opencv_python2=0` seems to have no effect. 😩

**Comment by patricksnape** [comment]
Weird because it works on my mac!

**Comment by jakirkham** [comment]
I can't believe I forgot `[skip appveyor]` again. Sorry.

Interesting, maybe my Mac is being weird. It kept showing that it picked up a Python 2 and Python 3 interpreter and both were the same.

**Comment by jakirkham** [comment]
I mean if you want to try this out I'm ok with that. I'm just not sure that this is enough yet.

**Comment by jakirkham** [comment]
Did this already get included? Seems like it has merge conflicts. I'll close it if we don't need it any more.

**Comment by patricksnape** [comment]
I was trying it yesterday - having problems with wget being missing?

**Pull request closed without merging**